services/community-site/src/pages/RunNode/Modal/EditValidator.tsx

```diff
@@ -50,21 +52,8 @@ const EditValidator = ({
     };
 
     if (type === 'mainnet') {
-      setLoading(true);
-      try {
-        const res = await setValidatorInfo(
-          validator.address,
-          payload.description,
-          payload.endpoint,
-        );
-        setStep(2);
-        await res.wait();
-        setLoading(false);
-      } catch (e) {
-        // eslint-disable-next-line no-console
-        console.error(e);
-        setLoading(false);
-      }
+      onSuccess();
+      asyncCallback(setValidatorInfo, [validator.address, payload.description, payload.endpoint]);

```

Comment:
Why not giving it a function as a param? Seems cleaner and simpler

Reply:
But I am. setValidatorInfo is the function, and the second param is the array of params the setValidatorInfo requires.

Reply:
Or maybe I don`t understand what you are suggesting

Reply:
I guess what Leo meant was a single parameter as such: `(args: any[]) => Promise<any>`

Reply:
With that, we could pass the methods we define on hooks directly into it like importing setValdiatorInfo() and then doing `asyncCallback(setValdiatorInfo([description, endpoint]))`

Reply:
Got it, changed